query/src/main/java/org/infinispan/query/dsl/embedded/impl/QueryEngine.java

```diff
@@ -666,15 +666,9 @@ protected RowProcessor makeProjectionProcessor(Class<?>[] projectedTypes, Object
                                             TimeoutExceptionFactory timeoutExceptionFactory,
                                             ExecutorService asyncExecutor,
                                             Class<?>... classes) {
-      CacheQuery cacheQuery;
-      if (indexedQueryMode == IndexedQueryMode.BROADCAST) {
-         cacheQuery = new ClusteredCacheQueryImpl<>(luceneQuery, getSearchFactory(), asyncExecutor, cache,
-               keyTransformationHandler, classes);
-      } else {
-         cacheQuery = new CacheQueryImpl<>(luceneQuery, getSearchFactory(), cache, keyTransformationHandler,
-               timeoutExceptionFactory, classes);
-      }
-      return (CacheQuery<E>) cacheQuery;
+      return new CacheQueryImpl<>(luceneQuery, getSearchFactory(), cache, keyTransformationHandler,
```

Comment:
So asyncExecutor and indexedQueryMode params are no longer used here. Why not remove them also from method signature? We can do that, QueryEngine is not API.